crates/sqlite-libsql/src/lib.rs

```diff
@@ -59,7 +61,7 @@ impl spin_sqlite::Connection for LibsqlClient {
         // As in `query`, the shenanigans just wrap a call to libsql's `Client::batch()`.
         std::thread::spawn(move || {
             let rt = tokio::runtime::Runtime::new().unwrap();
-            rt.block_on(client.batch(stmts))
+            rt.block_on(client.raw_batch(stmts))
```

Comment:
Do we know of any implications to changing from `batch` to `raw_batch` here?  (Apart from `batch` would require us to _further_ wrap our `http::Client` in a `client::Client`.)  The docs are a bit ambiguous.

I don't know if it matters but the comment on line 61 still refers to `batch()`.

Reply:
Thanks for asking this. I took another look and the `raw_batch` doesn't collect errors - so essentially we would swallow errors if the user's migration file was not valid. On top of that, we would not be running everything in a transaction which is also not good. I've switched back to using `batch` by wrapping the http client in a generic client.

This also made me look at some of the other code around and clean it up some hacks from the original implementation. This includes making the `Connection` trait async. This improves over what is currently in main since we'll now coordinate more appropriately with the tokio runtime.